test/test_torch.py

```diff
@@ -17489,6 +17493,52 @@ def test_hypot(self, device, dtype):
                 expected = np.hypot(input[0].cpu().numpy(), input[1].cpu().numpy())
             self.assertEqual(actual, expected)
 
+    @dtypesIfCPU(torch.bfloat16, torch.float32, torch.float64)
+    @dtypes(torch.float32, torch.float64)
+    @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
+    def test_igamma(self, device, dtype):
+        inputs = [
+            (torch.rand(10, device=device).to(dtype), torch.rand(10, device=device).to(dtype)),
+            (torch.rand((3, 3, 3), device=device).to(dtype), torch.rand((3, 3, 3), device=device).to(dtype)),
+            (torch.rand((100, 1), device=device).to(dtype), torch.rand((100, 1), device=device).to(dtype).transpose(0, 1)),
+            (torch.randint(100, (10, ), device=device, dtype=torch.long), torch.rand(10, device=device).to(dtype))
```

Comment:
How does this case work? I thought this function only supported float types?

Reply:
I like that you're programmatically enumerating the inputs and comparing with SciPy, but most of the tested values are in the range [0, 1). Are there other interesting ranges to probe for numerical accuracy?

Reply:
I think it converts the integer to float types. Should I just remove this?